kubemarine/plugins/calico.py

```diff
@@ -243,13 +243,24 @@ def enrich_deployment_calico_typha(cluster, obj_list):
     :param obj_list: list of objects for enrichment
     """
 
+    default_tolerations = [{'key': 'node.kubernetes.io/network-unavailable', 'effect': 'NoSchedule'},
+                           {'key': 'node.kubernetes.io/network-unavailable', 'effect': 'NoExecute'}]
+
     key = "Deployment_calico-typha"
     val = cluster.inventory['plugins']['calico']['typha']['replicas']
     obj_list[key]['spec']['replicas'] = int(val)
     cluster.log.verbose(f"The {key} has been patched in 'spec.replicas' with '{val}'")
     val = cluster.inventory['plugins']['calico']['typha']['nodeSelector']
     obj_list[key]['spec']['template']['spec']['nodeSelector'] = val
     cluster.log.verbose(f"The {key} has been patched in 'spec.template.spec.nodeSelector' with '{val}'")
+    if cluster.inventory['plugins']['calico']['typha']['tolerations']:
+        for val in default_tolerations:
+            obj_list[key]['spec']['template']['spec']['tolerations'].append(val)
+            cluster.log.verbose(f"The {key} has been patched in 'spec.template.spec.tolerations' with '{val}'")
+        vals = cluster.inventory['plugins']['calico']['typha']['tolerations']
```

Comment:
In current implementation we will insert default tolerations twice: from `default_tolerations` and from `defaults.yaml`. Was it intentional? Also, if we use `default_tolerations`, I think we need to insert them out of `if cluster.inventory['plugins']['calico']['typha']['tolerations']` condition.

Reply:
Oh, yes. I forgot about `defaults.yaml`